README.md

```diff
@@ -1,57 +1,122 @@
+| platform | type of sequencing platform, e.g. `NovaSeq` |
+| machine | specific machine id, e.g. NovaSeq instruments have `@Axxxxx` |
+| flowcell | identifer of flowcell used |
+| lane | flowcell lane number |
+| barcode | sequence library barcode/index, connect forward and reverse indices by `+`, e.g. `ATGC+ATGC` |
+| fastq1/2 | absolute path to forward and reverse reads |
+| adapter | adapter sequences to be trimmed, separated by comma |
 
 ### Reference data
 
-1. You need a ...
+You need have a indexed reference genome: ex reference.fna 
+
+For bwa the files are generated by [bwa index](http://bio-bwa.sourceforge.net/bwa.shtml#:~:text=SYNOPSIS-,bwa%20index%20ref.fa,-bwa%20mem%20ref):
+| File | Description |
+| -----| ----------- |
+| reference.dict | dictionary file|
+| reference.fna.amb | record appearance of N (or other non-ATGC) in the ref fasta |
+| reference.fna.ann | record ref sequences, name, length, etc |
+| reference.fna.bwt | the Burrows-Wheeler transformed sequence |
+| reference.fna.fai | index file|
+| reference.fna.pac | packaged sequence (four base pairs encode one byte) |
+| reference.fna.sa | suffix array index |
+
 
 ## :white_check_mark: Testing
 
 The workflow repository contains a small test dataset `.tests/integration` which can be run like so:
 
 ```bash
-cd .tests/integration
-snakemake -s ../../Snakefile -j1 --use-singularity
+$ cd .tests/integration
+$ snakemake -s ../../Snakefile -j1 --use-singularity
 ```
 
 ## :rocket: Usage
 
-The workflow is designed for WGS data meaning huge datasets which require a lot of compute power. For
-HPC clusters, it is recommended to use a cluster profile and run something like:
+To use this module in your workflow, follow the description in the
+[snakemake docs](https://snakemake.readthedocs.io/en/stable/snakefiles/modularization.html#modules).
+Add the module to your `Snakefile` like so:
 
 ```bash
-snakemake -s /path/to/Snakefile --profile my-awesome-profile
+module alignment:
+    snakefile:
+        github(
+            "hydra-genetics/alignment",
+            path="workflow/Snakefile",
+            tag="v0.1.0",
+        )
+    config:
+        config
+
+
+use rule * from alignment as alignment_*
 ```
 
+### Input files
+
+| File | Description |
+|---|---|
+| ***`hydra-genetics/prealignment data`*** |
+| `prealignment/fastp_pe/{sample}_{flowcell}_{lane}_{type}_fastq1.fastq.gz` | Merged and possibly trimmed foward reads |
+| `prealignment/fastp_pe/{sample}_{flowcell}_{lane}_{type}_fastq1.fastq.gz` | Merged and possibly trimmed reverse reads |
+| ***`original fastq files`*** |
+| `PATH/fastq1.fastq.gz` | forward reads retrieved from units.tsv |
+| `PATH/fastq2.fastq.gz` | reverse reads retrieved from units.tsv |
+
+
+### Output files
+
+The following output files should be targeted via another rule:
+
+| File | Description |
+|---|---|
+| `alignment/bwa_mem/{sample}_{type}.bam` | Aligned data |
```

Comment:
I think we should only indicate final output to make it less confusing?

Reply:
I will remove bwa_mem option! But I think the  "aligned data separated by chromosome" could make it easier for further development, if people want to start from chromosome seperated files.